controller/src/main/java/org/jboss/as/controller/registry/ImmutableManagementResourceRegistration.java

```diff
@@ -252,4 +253,11 @@
      * @return the set of ordered child types, and and empty set if there are none
      */
     Set<String> getOrderedChildTypes();
+
+    /**
+     * Returns capability definition for this resource.
+     *
+     * @return RuntimeCapability if any registered otherwise null
+     */
+    Set<RuntimeCapability<?>> getCapabilities();
```

Comment:
I think having just Capabilty would be fine, could even be new interface just to properly abstract this.